jetty-webapp/src/test/java/org/eclipse/jetty/webapp/WebAppClassLoaderTest.java

```diff
@@ -340,26 +335,4 @@ public void testResources() throws Exception
 
         assertThat("Resources Found (Parent Loader Priority == true) (with systemClasses filtering)", resources, ordered(expected));
     }
-
-    @Test
-    public void ordering() throws Exception
-    {
-        // The existence of a URLStreamHandler changes the behavior
-        assumeTrue(URLStreamHandlerUtil.getFactory() == null, "URLStreamHandler changes behavior, skip test");
-
-        Enumeration<URL> resources = _loader.getResources("org/acme/clashing.txt");
-        assertTrue(resources.hasMoreElements());
-        URL resource = resources.nextElement();
-        try (InputStream data = resource.openStream())
-        {
-            assertThat("correct contents of " + resource, IO.toString(data), is("alpha"));
-        }
-        assertTrue(resources.hasMoreElements());
-        resource = resources.nextElement();
-        try (InputStream data = resource.openStream())
-        {
-            assertThat("correct contents of " + resource, IO.toString(data), is("omega"));
-        }
-        assertFalse(resources.hasMoreElements());
-    }
```

Comment:
Why did this test get removed?

Reply:
Ordered elements within a classloader is not supported (by any classloader).
The fact that we never noticed this test failing is frankly pure luck.

Reply:
I belive it is testing the operation of WebAppClassLoader.addJars():  https://github.com/eclipse/jetty.project/blob/jetty-9.4.x/jetty-webapp/src/main/java/org/eclipse/jetty/webapp/WebAppClassLoader.java#L325

See https://github.com/eclipse/jetty.project/commit/ac61f0e968674728994c5855c5b5ad6c40441ed8

Reply:
This method has been restored, but adjusted to not test order of resources from `ClassLoader.getResources()` which isn't a features of a Java ClassLoader.
It merely tests that both clashing resources are returned, in any order.

Reply:
URLClassLoader explicitly supports ordering: https://github.com/openjdk/jdk/blob/master/src/java.base/share/classes/java/net/URLClassLoader.java#L90-L96
Even if this is an artefact of the implementation, we should keep the test to check that a solution we have provided continues to work.

Reply:
We don't use that that constructor for this test.
To get here, we wind up using URLClassLoader.addURL() based on a previously instantiated URLClassLoader.

The "previous solution" doesn't work, it has never worked reliably.

In this test case, the order is not determined by the the web descriptor.
It is determined by the order of the files being returned from asking for the list of files on the FileSystem.

It fails on OSX (on hfs+), it fails on Windows (with fat32, and ntfs), and it fails on some Linux filesystems (vfat, xfs, etc), but not on ext* based linux filesystems.
This test is totally bogus.
The order tests do exist, and those do work, check the [OrderingTest](https://github.com/eclipse/jetty.project/blob/jetty-9.4.x/jetty-webapp/src/test/java/org/eclipse/jetty/webapp/OrderingTest.java).